core/src/com/mygdx/game/scene/GameScene.java

```diff
@@ -256,12 +258,28 @@ public DogCharacter spawnDog(String sharedBlueprintId, Vector3 initialPosition)
 		return obj;
 	}
 
-	private <T extends SteerableBody> void setSteerableData(T obj) {
-		Ray posGroundRay = new Ray(obj.getPosition(), V3_DOWN);
-		obj.currentTriangle = navMesh.rayTest(posGroundRay, 100, null);
-		if (obj.currentTriangle == null) {
-			throw new GdxRuntimeException(String.format("Failed to find navigation mesh position for %s", obj));
-		}
+	/**
+	 * Finds the current triangle and sets the model to be visible on the same layer as mesh part index of current triangle
+	 * @param obj the steerable body
+	 */
+	public <T extends SteerableBody> void setSteerableData(T obj) {
+		final Vector3 pos = obj.getPosition();
+		final Ray posGroundRay = tmpRay.set(pos, V3_DOWN);
+		tmpRay.origin.y -= obj.halfExtents.y;
+//		final Triangle curTri = obj.currentTriangle;
+//		if (curTri == null || navMesh.getClosestPoint(curTri, tmpRay.origin, new Vector3()) > 1f) {
+//			Triangle newTriangle = navMesh.rayTest(posGroundRay, 100f, null);
+//			if (newTriangle == null) {
+//				if (obj.currentTriangle == null) {
+					Gdx.app.log(tag, String.format("Frame %d: Finding closest navigation mesh position for %s", Gdx.graphics.getFrameId(), obj));
+					obj.currentTriangle = navMesh.getClosestTriangle(pos, tmpRay.origin, null);
```

Comment:
It will work for now, I will try reducing the cost either by using built in methods for space partitioning in Bullet, or else write something for it myself at some point.